grails-app/services/org/pih/warehouse/product/AttributeService.groovy

```diff
@@ -4,13 +4,13 @@ import org.pih.warehouse.core.EntityTypeCode
 
 class AttributeService {
 
-    def list(EntityTypeCode entityTypeCode) {
+    def list(EntityTypeCode entityTypeCode, Boolean active = true) {
         String query = "select a from Attribute a"
         Map<String, String> argumentsList = [:]
 
         if (entityTypeCode) {
-            query += " join a.entityTypeCodes etc where etc = :entityTypeCode"
-            argumentsList += [entityTypeCode: entityTypeCode]
+            query += " join a.entityTypeCodes etc where etc = :entityTypeCode and active = :active"
+            argumentsList += [entityTypeCode: entityTypeCode, active: active]
```

Comment:
I wasn't sure how I should deal with the active property separately from entityTypeCode here, because the syntax of this query forced me to append the active in the where clause

Reply:
@alannadolny one minor thing for now, if you have above `select a from Attribute a`, and you use it in join as `join a.entityTypeCodes` then please use also `a` here as `and a.active = :active` for clarity.